src/main/java/uk/gov/ons/census/action/messaging/ActionFulfilmentReceiver.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.ons.census.action.messaging;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.integration.annotation.MessageEndpoint;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.census.action.model.dto.PrintFileDto;
+import uk.gov.ons.census.action.model.dto.ResponseManagementEvent;
+import uk.gov.ons.census.action.model.dto.UacQidDTO;
+import uk.gov.ons.census.action.model.entity.ActionType;
+import uk.gov.ons.census.action.model.entity.Case;
+import uk.gov.ons.census.action.model.repository.CaseRepository;
+import uk.gov.ons.census.action.service.CaseService;
+
+@MessageEndpoint
+public class ActionFulfilmentReceiver {
+  private static final Logger log = LoggerFactory.getLogger(ActionFulfilmentReceiver.class);
+  private final RabbitTemplate rabbitTemplate;
+  private final CaseService caseService;
+  private final CaseRepository caseRepository;
+
+  @Value("${queueconfig.outbound-exchange}")
+  private String outboundExchange;
+
+  @Value("${queueconfig.outbound-printer-routing-key}")
+  private String outboundPrinterRoutingKey;
+
+  public ActionFulfilmentReceiver(
+      RabbitTemplate rabbitTemplate, CaseService caseService, CaseRepository caseRepository) {
+    this.rabbitTemplate = rabbitTemplate;
+    this.caseService = caseService;
+    this.caseRepository = caseRepository;
+  }
+
+  @Transactional
+  @ServiceActivator(inputChannel = "actionFulfilmentInputChannel")
+  public void receiveEvent(ResponseManagementEvent responseManagementEvent) {
+    UUID caseId =
+        UUID.fromString(responseManagementEvent.getPayload().getFulfilmentRequest().getCaseId());
+    Optional<Case> fulfilmentCase = caseRepository.findByCaseId(caseId);
+    log.with("caseId", caseId).debug("Fulfilment Requested Event");
+    if (fulfilmentCase.isEmpty()) {
+      log.with("caseId", caseId).error("Cannot find Case for fulfilment request.");
+      String error = "Cannot find case for fulfilment request." + caseId;
+      throw new RuntimeException(error);
+    }
+    String packCode =
+        responseManagementEvent.getPayload().getFulfilmentRequest().getFulfilmentCode();
+    Optional<Integer> questionnaireType = determineQuestionnaireType(packCode);
+    if (questionnaireType.isEmpty()) {
+      log.with("caseId", caseId)
+          .with("PackCode", packCode)
+          .error("Unknown packCode in fulfilment request.");
+      String error = "Unknown packCode in fulfilment request. " + packCode + " caseId " + caseId;
+      throw new RuntimeException(error);
+    }
+    UacQidDTO uacQid = caseService.getUacQid("1");
```

Comment:
What does "1" mean?

Reply:
Presumably this has been hard-coded as "1" when it should have been `questionnaireType.get()`

Reply:
Fixed.